### PR TITLE
Switch description from query parameter to form data sending deploy description to dag server

### PR DESCRIPTION
**Pull request by Simpcyclassy**
## Description

This PR modifies the way the `description` is handled in the `UploadFile` function. Previously, the `description` was passed as a query parameter in the URL. It is now switched to be sent as form data along with the file in the multipart request.

### Key changes:
- Switched `description` from a query parameter to form data.
- Added the `description` field to the `UploadFileArguments` struct.
- Updated the `UploadFile` function to handle the `description` as form data.

This change improves the handling of spaces and special characters in the `description`, avoiding issues caused by URL encoding.

Related astronomer/issues/issues/6663

## 🧪 Functional Testing

- Adjusted all related test cases in `TestUploadFile` to reflect this change.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [x] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: pkg/fileutil/files_test.go [hunk 1]
```diff
@@ -814,6 +820,7 @@ func (s *Suite) TestUploadFile() {
 			TargetURL:           server.URL,
 			FormFileFieldName:   "file",
 			Headers:             headers,
+			Description:         "This is a test description",
```
Comment (jeremybeard) [inline pkg/fileutil/files_test.go:823]: Can we update this test case to check the contents of the request, to verify the file contents and the description are in there?

File: pkg/fileutil/files.go [hunk 2]
```diff
@@ -326,9 +334,8 @@ func UploadFile(args *UploadFileArguments) error {
 		}
 
 		// set headers
-		for k, v := range headers {
-			req.Header.Set(k, v)
-		}
+		setHeaders(req, headers)
+
 		client := &http.Client{}
 		response, err := client.Do(req)
```
Comment (Simpcyclassy) [inline pkg/fileutil/files.go:340]: I only tested for no description. I was unable to fully test the description. When logging the request body just before and after this line `response, err := client.Do(req)`, I noticed that no data is logged after the request is sent.
Given this, simulating and testing errors for the description field and even file did not quite work.

File: pkg/fileutil/files_test.go [hunk 3]
```diff
@@ -814,6 +820,7 @@ func (s *Suite) TestUploadFile() {
 			TargetURL:           server.URL,
 			FormFileFieldName:   "file",
 			Headers:             headers,
+			Description:         "Deployed via <astro deploy --dags>n",
```
Comment (neel-astro) [inline pkg/fileutil/files_test.go:823]: Might want to validate that the request body contains all the required elements and is in the correct format for this test (similar to the previous comment from Jeremy).

**Review by neel-astro — COMMENTED** [review]
Bringing in @lzdanski's comment from the docs PR for release changes, could we update the description of the newly added `description` flag, to match the description added in docs ([here](https://github.com/astronomer/docs/pull/4231/files#diff-281913e3f18f7588b9bd697ad8994d5c2be26b0114f278241fff0cc5134e278dR118)) for consistency.

**Review by neel-astro — APPROVED** [review]

**Comment by Simpcyclassy** [comment]
I do not have the permission to merge. I need help merging this @neel-astro

**Pull request merged**